pennylane_lightning/core/src/simulators/lightning_gpu/bindings/LGPUBindings.hpp

```diff
@@ -384,9 +384,20 @@ void registerBackendSpecificInfo(py::module_ &m) {
         .def("acquireDevice", &DevicePool<int>::acquireDevice)
         .def("releaseDevice", &DevicePool<int>::releaseDevice)
         .def("syncDevice", &DevicePool<int>::syncDevice)
+        .def("refresh", &DevicePool<int>::refresh)
         .def_static("getTotalDevices", &DevicePool<int>::getTotalDevices)
         .def_static("getDeviceUIDs", &DevicePool<int>::getDeviceUIDs)
-        .def_static("setDeviceID", &DevicePool<int>::setDeviceIdx);
+        .def_static("setDeviceID", &DevicePool<int>::setDeviceIdx)
+        .def(py::pickle(
+            [](const DevicePool<int> &self) { // __getstate__
+                return py::make_tuple();
+            },
+            [](py::tuple &t) { // __setstate__
+                if (t.size() != 0) {
+                    throw std::runtime_error("Invalid state!");
+                }
```

Comment:
Where is this tested? Why is this condition required?

Reply:
I am not sure what the point of this condition is, It was in the original implementation by @mlxd.

Reply:
This is part of the requirements from the Pybind11 serialisation --- this is never hit by anything but pybind11 internals, but I added safety catches in the event of a change. See https://github.com/pybind/pybind11/blob/5552cbf2058d9e42182287f0b3ebc86066cb9b7a/tests/test_pickling.cpp#L45 for an example from their tests

Reply:
Gotta! Then I may suggest using PL_Abort to get a more descriptive error message here instead of `std::runtime_error`:
```suggestion
                PL_ABORT_IF(t.size(), "Invalid state!");
```